components/cam/src/physics/cam/shoc.F90

```diff
@@ -1812,10 +1813,13 @@ subroutine x_y_terms_diag_third_shoc_moment(&
   !intent-outs
   real(rtype), intent(out) :: x0, y0, x1, y1
 
-  real(rtype), parameter :: a0=(0.52_rtype*c_diag_3rd_mom**(-2))/(c_diag_3rd_mom-2._rtype)
-  real(rtype), parameter :: a1=0.87_rtype/(c_diag_3rd_mom**2)
-  real(rtype), parameter :: a2=0.5_rtype/c_diag_3rd_mom
-  real(rtype), parameter :: a3=0.6_rtype/(c_diag_3rd_mom*(c_diag_3rd_mom-2._rtype))
+  ! local variables
+  real(rtype) :: a0, a1, a2, a3
+
+  a0 = (0.52_rtype*bfb_pow(c_diag_3rd_mom,-2._rtype))/(c_diag_3rd_mom-2._rtype)
```

Comment:
Maybe this changed, but I believe bfb_square, etc, are macros, so you can keep these as parameters.

Reply:
They are still macros.

Reply:
I initially tried this, but I got a `Function ‘cxx_pow’ in initialization expression at (1) must be an intrinsic function` error.

Reply:
Yeah, cxx_pow requires a cuda kernel launch, so for sure not good for parameter in F90. But even on host, cxx_pow is not something that can be compiled away (it comes from another module).

And the compiler error seems to suggest it would not work even if cxx_pow was defined in the same module.

I guess no constexpr in F90...

Reply:
Ok, yeah, can't do it b/c of the C++ code path. I got that wrong in my initial comment.